### PR TITLE
Multiple Log Bucket Locations

### PR DESCRIPTION
**Pull request by matt-heery**
Allows to have three separate ways of defining log bucket location:
log_buckets, needs to be a map of s3 buckets to log to eg `tomap({"first_log_bucket: module.s3-log-bucket-1.bucket, "second_log_bucket: module.s3-log-bucket-2bucket})`
log_bucket_names, needs to be names of s3 buckets (not defined in the same terraform config) eg `toset(["test-bucket-name-1", "test-bucket-name-2" ])`
or log_bucket being: `"test-bucket-name-1"`

### CODE REVIEW

File: test/unit-test/main.tf [hunk 1]
```diff
@@ -58,6 +58,45 @@ module "s3_with_notification" {
...
+#trivy:ignore:AVD-AWS-0132
+resource "aws_s3_bucket" "non-modulised-bucket" {
+  #checkov:skip=CKV2_AWS_6: "Ensure that S3 bucket has a Public Access block - This is not needed in our tests"
+  #checkov:skip=CKV_AWS_18: "Ensure the S3 bucket has access logging enabled - This is not needed in our tests"
+  #checkov:skip=CKV_AWS_21: "Ensure all data stored in the S3 bucket have versioning enabled - This is not needed in our tests"
+  #checkov:skip=CKV2_AWS_61: "Ensure that an S3 bucket has a lifecycle configuration - This is not needed in our tests"
+  #checkov:skip=CKV2_AWS_62: "Ensure S3 buckets should have event notifications enabled - This is not needed in our tests"
+  #checkov:skip=CKV_AWS_144: "Ensure that S3 bucket has cross-region replication enabled - This is not needed in our tests"
+  #checkov:skip=CKV_AWS_145: "Ensure that S3 buckets are encrypted with KMS by default - This is not needed in our tests"
+  bucket = "log-test-bucket-051683332738327"
```
Comment (dms1981) [inline test/unit-test/main.tf:77]: `bucket_prefix = "log-test-bucket"` would keep resources here unique which might be better for unit tests?

File: test/unit-test/main.tf [hunk 2]
```diff
@@ -58,6 +58,45 @@ module "s3_with_notification" {
...
+resource "aws_s3_bucket" "non-modulised-bucket-2" {
+  #checkov:skip=CKV2_AWS_6: "Ensure that S3 bucket has a Public Access block - This is not needed in our tests"
+  #checkov:skip=CKV_AWS_18: "Ensure the S3 bucket has access logging enabled - This is not needed in our tests"
+  #checkov:skip=CKV_AWS_21: "Ensure all data stored in the S3 bucket have versioning enabled - This is not needed in our tests"
+  #checkov:skip=CKV2_AWS_61: "Ensure that an S3 bucket has a lifecycle configuration - This is not needed in our tests"
+  #checkov:skip=CKV2_AWS_62: "Ensure S3 buckets should have event notifications enabled - This is not needed in our tests"
+  #checkov:skip=CKV_AWS_144: "Ensure that S3 bucket has cross-region replication enabled - This is not needed in our tests"
+  #checkov:skip=CKV_AWS_145: "Ensure that S3 buckets are encrypted with KMS by default - This is not needed in our tests"
+
+  bucket = "log-test-bucket-2-051683332738327"
```
Comment (dms1981) [inline test/unit-test/main.tf:97]: Same here - `bucket_prefix` over a static name.

File: test/unit-test/main.tf [hunk 3]
```diff
@@ -75,10 +114,13 @@ module "s3_with_log_bucket" {
...
     aws.bucket-replication = aws
   }
-  bucket_prefix = "unit-test-bucket-with-logs"
-  force_destroy = true
-  log_buckets   = tomap({ "main_log_bucket" : module.dummy_s3_log_bucket.bucket })
-  tags          = local.tags
+  bucket_prefix    = "unit-test-bucket-with-logs"
+  force_destroy    = true
+  log_buckets      = tomap({ "main_log_bucket" : module.dummy_s3_log_bucket.bucket })
+  log_bucket_names = toset(["log-test-bucket-051683332738327"])
```
Comment (dms1981) [inline test/unit-test/main.tf:120]: Would this be better with multiple buckets to test the behaviour?

File: test/unit-test/main.tf (hunk 1 above)
Comment (matt-heery) [reply test/unit-test/main.tf:77]: i tried this originally, but the problem is (i forgot it was the original problem i was tackling) if i try and call the bucket name via code in the same terraform plan: it fails (as there's a conditional on something that hasnt been generated yet), so this was the only way i could get it to work 😓

File: test/unit-test/main.tf (hunk 3 above)
Comment (matt-heery) [reply test/unit-test/main.tf:120]: yeah good point - will wait to resolve the above before adding another bucket :)

File: test/unit-test/main.tf (hunk 1 above)
Comment (dms1981) [reply test/unit-test/main.tf:77]: OK! Makes total sense - thanks for the explanation :)

**Review by dms1981 — APPROVED** [review]

**Pull request merged**